pkg/signal/signal.go

```diff
@@ -116,7 +120,7 @@ func UnmuteChannel(signal int) error {
 }
 
 // RegisterChannel registers a go channel for a given signal.
-func RegisterChannel(signal int, ch chan<- int) error {
```

Comment:
If this function is planned to be used concurrently perhaps it would be better to protect `channels` against concurrent usage.

Reply:
I wouldn't overdesign it at this point, only once we add more signals from BPF datapath to it and such need arises as a result of it, but as it stands right now it is fully sufficient to register everything beforehand and then run the go thread.